exonum-java-binding-core/rust/src/proxy/service.rs

```diff
@@ -75,15 +75,16 @@ impl Service for ServiceProxy {
     fn state_hash(&self, snapshot: &Snapshot) -> Vec<Hash> {
         unwrap_jni(self.exec.with_attached(|env| {
             let view_handle = to_handle(View::from_ref_snapshot(snapshot));
-            let java_service_hashes = panic_on_exception(
-                env,
-                env.call_method(
+            let java_service_hashes = panic_on_exception(env, unsafe {
+                env.call_method_unsafe(
                     self.service.as_obj(),
-                    "getStateHashes",
-                    "(J)[[B",
+                    jni_cache::get_usa_state_hashes(),
+                    JavaType::Array(Box::new(JavaType::Array(Box::new(JavaType::Primitive(
```

Comment:
Can anyone say why jni-rs requires Box for Array enum variant?

Reply:
… but, hurts the readability of the thing. 

Also, some of the examples in https://github.com/jni-rs/jni-rs/issues/137 are not correct :smile: